FlowCal/io.py

```diff
@@ -164,21 +164,18 @@ def read_fcs_text_segment(buf, begin, end, delim=None):
     buf.seek(begin)
     raw = buf.read((end+1)-begin)
 
-    # Some files have been found to include space characters at the end of the
-    # TEXT segment. If the delimiter is not a space character, strip spaces
-    # from the end of the segment.
-    if not delim.isspace():
-        raw = raw.rstrip()
+    # Check that the first character of the TEXT segment is equal to the
+    # delimiter.
+    if raw[0] != delim:
+        raise ValueError("segment should start with delimiter")
 
-    pairs_list = raw.split(delim)
+    # Look for the last delimiter in the segment string, and retain everything
+    # from one character after the first delimiter to one character before the
+    # last delimiter.
+    end_index = raw.rfind(delim)
+    raw = raw[1: end_index]
```

Comment:
The edges cases I can think of for this appear to work in an intuitive manner (assuming `delim='/'`):

1. `raw=''` => `end_index=-1` => new `raw=''` => `pairs_list=['']`
2. `raw='/'` => `end_index=0` => new `raw=''` => `pairs_list=['']`
3. `raw='//'` => `end_index=1` => new `raw=''` => `pairs_list=['']`
4. `raw='///'` => `end_index=2` => new `raw='/'` => `pairs_list=['','']`

I think 1-3 will fail later due to:

```python
# List length should be even since all key-value entries should be pairs
if len(pairs_list) % 2 != 0:
    raise ValueError("odd # of (keys + values); unpaired key or value")
```

Could possibly throw an explicit error message for these, but not sure they warrant their own message.